File: AplicacionPeliculas/ModificaGenero.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;

namespace AplicacionPeliculas
{
    public partial class ModificaGenero
    {
        public int idgenero;
        public string nombregenero;
        private string query;

        public ModificaGenero()
        {
            InitializeComponent();
        }
        private void DataModgenero_CellContentClick(object sender, DataGridViewCellEventArgs e)
        {
            try
            {
                idgenero = Conversions.ToInteger(DataModgenero.Rows[e.RowIndex].Cells[0].Value);
                nombregenero = Conversions.ToString(DataModgenero.Rows[e.RowIndex].Cells[1].Value);
                My.MyProject.Forms.Genero.txtidgen.Text = idgenero.ToString();
                My.MyProject.Forms.Genero.txtNomGenero.Text = nombregenero;
                My.MyProject.Forms.Genero.Show();
                Close();
            }
            catch (Exception ex)
            {
                Interaction.MsgBox("Error" + Constants.vbCrLf + "Haz doble clic sobre el nombre de la película a buscar.");
            }
        }
        public void flitrarIdGen()
        {
            query = "select * from Genero where id LIKE '%" + txtidGen.Text + "%' and estatus = 1";
            DataModgenero.DataSource = Conecction.SelectQuery(query);
        }

        public void flitrarGen()
        {
            query = "select * from Genero where nombre LIKE '%" + txtNomGen.Text + "%' and estatus = 1";
            DataModgenero.DataSource = Conecction.SelectQuery(query);
        }

        private void txtNomGen_TextChanged(object sender, EventArgs e)
        {
            flitrarGen();
        }

        private void txtidGen_TextChanged(object sender, EventArgs e)
        {
            flitrarIdGen();
        }

    }
}

File: AplicacionPeliculas/ModificaGenero.vb
Public Class ModificaGenero
    Public idgenero As Integer
    Public nombregenero As String
    Dim query As String
    Private Sub DataModgenero_CellContentClick(sender As Object, e As DataGridViewCellEventArgs) Handles DataModgenero.CellContentClick
        Try
            idgenero = DataModgenero.Rows(e.RowIndex).Cells(0).Value
            nombregenero = DataModgenero.Rows(e.RowIndex).Cells(1).Value
            Genero.txtidgen.Text = idgenero
            Genero.txtNomGenero.Text = nombregenero
            Genero.Show()
            Me.Close()
        Catch ex As Exception
            MsgBox("Error" + vbCrLf + "Haz doble clic sobre el nombre de la película a buscar.")
        End Try
    End Sub
    Sub flitrarIdGen()
        query = "select * from Genero where id LIKE '%" & txtidGen.Text & "%' and estatus = 1"
        DataModgenero.DataSource = Conecction.SelectQuery(query)
    End Sub

    Sub flitrarGen()
        query = "select * from Genero where nombre LIKE '%" & txtNomGen.Text & "%' and estatus = 1"
        DataModgenero.DataSource = Conecction.SelectQuery(query)
    End Sub

    Private Sub txtNomGen_TextChanged(sender As Object, e As EventArgs) Handles txtNomGen.TextChanged
        flitrarGen()
    End Sub

    Private Sub txtidGen_TextChanged(sender As Object, e As EventArgs) Handles txtidGen.TextChanged
        flitrarIdGen()
    End Sub

End Class